src/dmd/root/object.h

```diff
@@ -57,6 +58,9 @@ class RootObject
     virtual void print();
 
     virtual const char *toChars();
+    /// This function is `extern(D)` and should not be called from C++,
+    /// as the ABI does not match on some platforms
+    virtual DArray<const char> toString();
```

Comment:
Is it possible to implement this on the C++ and have the implementation assert?

Reply:
I don't think so.